pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -1225,19 +1225,10 @@ func ExtractMsgDetails(msg *dns.Msg) (
 	}
 	qname = strings.ToLower(string(msg.Question[0].Name))
 
-	// rrName is the name the next RR should include.
-	// This will change when we see CNAMEs.
-	rrName := strings.ToLower(qname)
-
 	TTL = math.MaxUint32 // a TTL must exist in the RRs
 
 	answerTypes = make([]uint16, 0, len(msg.Answer))
 	for _, ans := range msg.Answer {
-		// Ensure we have records for DNS names we expect
-		if strings.ToLower(ans.Header().Name) != rrName {
```

Comment:
Do we want to log a warning(debug) atleast if this is the case ? (Thinking from the debugging perspective of cilium)